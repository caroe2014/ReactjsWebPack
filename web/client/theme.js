// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
