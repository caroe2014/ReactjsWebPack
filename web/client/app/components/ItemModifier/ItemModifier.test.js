// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {shallow, mount, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';
import ItemModifier from './ItemModifier';
import jsdom from 'jsdom';

configure({
  adapter: new Adapter()
});

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  },
}));

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

global.window = new jsdom.JSDOM().window;
global.document = window.document;
/* global describe, it, expect, beforeEach */

describe('ItemModifier', () => {

  let mockFn = jest.fn();
  let wrapper;
  let restProps;
  let mockProps = {
    modifiers: [
      {
        id: 'bread',
        description: 'BREAD',
        type: 'radio',
        minimum: 5,
        options: [
          {
            id: 'ciabatta',
            description: 'Ciabatta',
            selected: false,
            amount: '20'
          },
          {
            id: 'sourdough',
            description: 'Sourdough',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'marblerye',
            description: 'MarbleRye',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'multigrain',
            description: 'Multi Grain',
            selected: false
          },
          {
            id: 'wheatbaguette',
            description: 'Wheat Baguette',
            selected: false
          },
          {
            id: 'wheatwrap',
            description: 'Wheat wrap',
            selected: false
          },
          {
            id: 'spinachwrap',
            description: 'Spinach wrap',
            selected: false
          }
        ]
      },
      {
        id: 'proteins',
        description: 'PROTEINS',
        type: 'checkbox',
        minimum: 2,
        options: [
          {
            id: 'roastedbeef',
            description: 'Roatedbeef',
            selected: false
          },
          {
            id: 'bakedham',
            description: 'Baked Ham',
            selected: false
          },
          {
            id: 'roastedturkey',
            description: 'Roasted Turkey',
            selected: false
          },
          {
            id: 'multigrain',
            description: ' Multi Grain',
            selected: false
          },
          {
            id: 'cheickenbreast',
            description: 'Chicken Breast',
            selected: false
          },
          {
            id: 'bacon',
            description: 'Bacon',
            selected: false
          },
          {
            id: 'Chicken Salad',
            description: 'Chicken Salad',
            selected: false
          }
        ]
      },
      {
        id: 'addon',
        description: 'ADD ON',
        type: 'checkbox',
        minimum: 1,
        options: [
          {
            id: 'Fountain Drink',
            description: 'Fountain Drink',
            selected: false,
            amount: '2.50',
            options: [{
              id: 'Large',
              description: 'Large',
              selected: true,
              amount: '2.50'
            },
            {
              id: 'Small',
              description: 'Small',
              selected: false,
              amount: '1.50'
            }]
          },
          {
            id: 'Small Soup',
            description: 'Small Soup',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'Large Soup',
            description: 'Aged Cheddar',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'Swiss',
            description: 'Swiss',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'American',
            description: 'American',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'Mozarella',
            description: 'Mozarella',
            selected: false,
            amount: '1.50'
          },
          {
            id: 'No cheese',
            description: 'No cheese',
            selected: false,
            amount: '1.50'
          }
        ]
      }
    ],
    addOnAmount: 0
  };
  restProps = {
    showValidationState: 'test'
  };

  beforeEach(() => {
    wrapper = shallow(<ItemModifier modifiers={mockProps} {...restProps}/>
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    const mountWrapper = mount(<ThemeProvider theme={theme}><ItemModifier modifiers={mockProps} {...restProps}/></ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

  describe('when the page gets loaded', () => {
    it('should select first tab by default', () => {
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[0].id);
    });
  });

  describe('when a tab is selected', () => {
    it('should select the appropriate modifier', () => {
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[0].id);
      const input = wrapper.find('.' + mockProps.modifiers[1].id);
      input.simulate('click');
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[1].id);
    });
  });

  describe('when a tab is selected', () => {
    it('should select the appropriate modifier', () => {
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[0].id);
      const input = wrapper.find('.' + mockProps.modifiers[2].id);
      input.simulate('click');
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[2].id);
    });
  });

  describe('when a tab is selected', () => {
    it('should display its options', () => {
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[0].id);
      const input = wrapper.find('.' + mockProps.modifiers[1].id);
      input.simulate('click');
      expect(wrapper.state().selectedTab).toBe(mockProps.modifiers[1].id);
      expect(wrapper.find('.option').length).toBe(mockProps.modifiers[1].options.length);
    });
  });

  describe('when a modifier is selected', () => {
    it('should update the appropriate modifier', () => {
      let shalloWrapper = shallow(<ItemModifier modifiers={mockProps} handlerFromParent={mockFn} {...restProps}/>);
      const spy = jest.spyOn(shalloWrapper.instance(), 'updateModifierOption');
      shalloWrapper.instance().forceUpdate();
      shalloWrapper.find(`.option_id_${mockProps.modifiers[0].options[0].id}`).simulate('click');
      expect(spy).toHaveBeenCalled();
      spy.mockClear();
    });
  });

  describe('when a modifier is selected on mount', () => {
    let newMockProps = {
      modifiers: [
        {
          id: 'addon',
          description: 'ADD ON',
          type: 'checkbox',
          minimum: 1,
          options: [
            {
              id: 'Fountain Drink',
              description: 'Fountain Drink',
              selected: false,
              amount: '2.50',
              maximum: 10,
              options: [{
                id: 'Large',
                description: 'Large',
                selected: true,
                amount: '2.50'
              },
              {
                id: 'Small',
                description: 'Small',
                selected: false,
                amount: '1.50'
              }]
            },
            {
              id: 'Small Soup',
              description: 'Small Soup',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Large Soup',
              description: 'Aged Cheddar',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Swiss',
              description: 'Swiss',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'American',
              description: 'American',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Mozarella',
              description: 'Mozarella',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'No cheese',
              description: 'No cheese',
              selected: false,
              amount: '1.50'
            }
          ]
        }
      ],
      addOnAmount: 0
    };

    it('should update the appropriate modifier', () => {
      let mountWrapper = mount(<ThemeProvider theme={theme}><ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/></ThemeProvider>);
      expect(mountWrapper.exists()).toBe(true);
    });
  });

  describe('when a modifier minimum is 0', () => {
    let newMockProps = {
      modifiers: [
        {
          id: 'addon',
          description: 'ADD ON',
          type: 'checkbox',
          minimum: 0,
          options: [
            {
              id: 'Fountain Drink',
              description: 'Fountain Drink',
              selected: false,
              amount: '2.50',
              maximum: 10,
              options: [{
                id: 'Large',
                description: 'Large',
                selected: true,
                amount: '2.50'
              },
              {
                id: 'Small',
                description: 'Small',
                selected: false,
                amount: '1.50'
              }]
            },
            {
              id: 'Small Soup',
              description: 'Small Soup',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Large Soup',
              description: 'Aged Cheddar',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Swiss',
              description: 'Swiss',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'American',
              description: 'American',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Mozarella',
              description: 'Mozarella',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'No cheese',
              description: 'No cheese',
              selected: false,
              amount: '1.50'
            }
          ]
        }
      ],
      addOnAmount: 0
    };

    it('should update the appropriate modifier', () => {
      let shallowWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      expect(shallowWrapper.exists()).toBe(true);
    });
  });

  describe('when a modifier max is 1', () => {
    let newMockProps = {
      modifiers: [
        {
          id: 'addon',
          description: 'ADD ON',
          type: 'checkbox',
          minimum: 1,
          maximum: 1,
          options: [
            {
              id: 'Fountain Drink',
              description: 'Fountain Drink',
              selected: false,
              amount: '2.50',
              type: 'checkbox',
              maximum: 10,
              options: [{
                id: 'Large',
                description: 'Large',
                selected: true,
                amount: '2.50'
              },
              {
                id: 'Small',
                description: 'Small',
                selected: false,
                amount: '1.50'
              }]
            },
            {
              id: 'Small Soup',
              description: 'Small Soup',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Large Soup',
              description: 'Aged Cheddar',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Swiss',
              description: 'Swiss',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'American',
              description: 'American',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Mozarella',
              description: 'Mozarella',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'No cheese',
              description: 'No cheese',
              selected: false,
              amount: '1.50'
            }
          ]
        }
      ],
      addOnAmount: 0
    };

    it('should update the appropriate modifier', () => {
      let shallowWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      expect(shallowWrapper.exists()).toBe(true);
    });

    it('should update the appropriate modifier', () => {
      let shalloWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      const spy = jest.spyOn(shalloWrapper.instance(), 'updateSubOption');
      shalloWrapper.instance().forceUpdate();
      shalloWrapper.find(`.subitem_${newMockProps.modifiers[0].options[0].options[0].id}`).simulate('click');
      expect(spy).toHaveBeenCalled();
      spy.mockClear();
    });

  });

  describe('when a modifier has type radio', () => {
    let newMockProps = {
      modifiers: [
        {
          id: 'addon',
          description: 'ADD ON',
          type: 'radio',
          minimum: 1,
          maximum: 1,
          options: [
            {
              id: 'Fountain Drink',
              description: 'Fountain Drink',
              selected: false,
              amount: '2.50',
              type: 'radio',
              maximum: 10,
              options: [{
                id: 'Large',
                description: 'Large',
                selected: true,
                amount: '2.50'
              },
              {
                id: 'Small',
                description: 'Small',
                selected: false,
                amount: '1.50'
              }]
            },
            {
              id: 'Small Soup',
              description: 'Small Soup',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Large Soup',
              description: 'Aged Cheddar',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Swiss',
              description: 'Swiss',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'American',
              description: 'American',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Mozarella',
              description: 'Mozarella',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'No cheese',
              description: 'No cheese',
              selected: false,
              amount: '1.50'
            }
          ]
        }
      ],
      addOnAmount: 0
    };

    it('should update the appropriate modifier', () => {
      let shallowWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      expect(shallowWrapper.exists()).toBe(true);
    });

    it('should update the appropriate modifier', () => {
      let shalloWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      const spy = jest.spyOn(shalloWrapper.instance(), 'updateSubOption');
      shalloWrapper.instance().forceUpdate();
      shalloWrapper.find(`.subitem_${newMockProps.modifiers[0].options[0].options[0].id}`).simulate('click');
      expect(spy).toHaveBeenCalled();
      spy.mockClear();
      shalloWrapper.unmount();
    });

  });

  describe('when a modifier is selected for radio', () => {
    let newMockProps = {
      modifiers: [
        {
          id: 'addon',
          description: 'ADD ON',
          type: 'radio',
          minimum: 1,
          maximum: 1,
          options: [
            {
              id: 'Fountain Drink',
              description: 'Fountain Drink',
              selected: false,
              amount: '2.50',
              type: 'radio',
              maximum: 10,
              options: [{
                id: 'Large',
                description: 'Large',
                selected: true,
                amount: '2.50'
              },
              {
                id: 'Small',
                description: 'Small',
                selected: false,
                amount: '1.50'
              }]
            },
            {
              id: 'Small Soup',
              description: 'Small Soup',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Large Soup',
              description: 'Aged Cheddar',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Swiss',
              description: 'Swiss',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'American',
              description: 'American',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Mozarella',
              description: 'Mozarella',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'No cheese',
              description: 'No cheese',
              selected: false,
              amount: '1.50'
            }
          ]
        }
      ],
      addOnAmount: 0
    };
    it('should update the appropriate modifier', () => {
      let shalloWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      const spy = jest.spyOn(shalloWrapper.instance(), 'updateModifierOption');
      shalloWrapper.instance().forceUpdate();
      shalloWrapper.find('.option_id_Fountain').simulate('click');
      expect(spy).toHaveBeenCalled();
      spy.mockClear();
    });
  });

  describe('when a modifier is selected for checkbox', () => {
    let newMockProps = {
      modifiers: [
        {
          id: 'addon',
          description: 'ADD ON',
          type: 'checkbox',
          minimum: 1,
          maximum: 1,
          options: [
            {
              id: 'Fountain Drink',
              description: 'Fountain Drink',
              selected: false,
              amount: '2.50',
              type: 'checkbox',
              maximum: 10,
              options: [{
                id: 'Large',
                description: 'Large',
                selected: true,
                amount: '2.50'
              },
              {
                id: 'Small',
                description: 'Small',
                selected: false,
                amount: '1.50'
              }]
            },
            {
              id: 'Small Soup',
              description: 'Small Soup',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Large Soup',
              description: 'Aged Cheddar',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Swiss',
              description: 'Swiss',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'American',
              description: 'American',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'Mozarella',
              description: 'Mozarella',
              selected: false,
              amount: '1.50'
            },
            {
              id: 'No cheese',
              description: 'No cheese',
              selected: false,
              amount: '1.50'
            }
          ]
        }
      ],
      addOnAmount: 0
    };
    it('should update the appropriate modifier', () => {
      let shalloWrapper = shallow(<ItemModifier modifiers={newMockProps} handlerFromParent={mockFn} {...restProps}/>);
      const spy = jest.spyOn(shalloWrapper.instance(), 'updateModifierOption');
      shalloWrapper.instance().forceUpdate();
      shalloWrapper.find('.option_id_Fountain').simulate('click');
      expect(spy).toHaveBeenCalled();
      spy.mockClear();
    });
  });

});
