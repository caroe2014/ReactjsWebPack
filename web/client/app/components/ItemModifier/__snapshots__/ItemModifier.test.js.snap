// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemModifier does a snapshot test 1`] = `
<ThemeProvider
  className="item-modifier-block"
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <styled.div
    className="modifiers"
    innerRef={[Function]}
    max-width={0}
  >
     
    <styled.div
      className="modifier-tabs"
      innerRef={[Function]}
      role="tablist"
      tabIndex={0}
    >
      <ArrowHorizontalScroll
        classContext="modifier-tabs-list"
        contentStyle={
          Object {
            "minHeight": "60px",
            "width": "100%",
          }
        }
        leftArrow={
          <Styled(Styled(Flex))
            tabIndex={0}
          >
            <i
              className="fa fa-angle-left"
            />
          </Styled(Styled(Flex))>
        }
        rightArrow={
          <Styled(Styled(Flex))
            tabIndex={0}
          >
            <i
              className="fa fa-angle-right"
            />
          </Styled(Styled(Flex))>
        }
      >
        <Styled(Tab)
          aria-label="BREAD ITEM_DETAILS_REQUIRED"
          aria-selected={true}
          className="tab_1 bread active"
          id="tab-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
          tabIndex={0}
        >
          <Styled(Text)>
            BREAD
          </Styled(Text)>
          <Styled(Text)
            className="required-text"
            error="test"
          >
            <div
              i18nKey="ITEM_DETAILS_REQUIRED"
            />
          </Styled(Text)>
        </Styled(Tab)>
        <styled.div />
        <Styled(Tab)
          aria-label="PROTEINS ITEM_DETAILS_REQUIRED"
          aria-selected={false}
          className="tab_2 proteins "
          id="tab-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
          tabIndex={0}
        >
          <Styled(Text)>
            PROTEINS
          </Styled(Text)>
          <Styled(Text)
            className="required-text"
            error="test"
          >
            <div
              i18nKey="ITEM_DETAILS_REQUIRED"
            />
          </Styled(Text)>
        </Styled(Tab)>
        <styled.div />
        <Styled(Tab)
          aria-label="ADD ON ITEM_DETAILS_REQUIRED"
          aria-selected={false}
          className="tab_3 addon "
          id="tab-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
          tabIndex={0}
        >
          <Styled(Text)>
            ADD ON
          </Styled(Text)>
          <Styled(Text)
            className="required-text"
            error="test"
          >
            <div
              i18nKey="ITEM_DETAILS_REQUIRED"
            />
          </Styled(Text)>
        </Styled(Tab)>
        <styled.div />
      </ArrowHorizontalScroll>
    </styled.div>
    <Styled(Flex)>
      <Styled(Flex)
        aria-label="ITEM_MODIFIERS_ADDON_MIN_MAX"
        className="header-text"
        id="modifiers-choose"
        tabIndex={0}
      >
        ITEM_MODIFIERS_ADDON_MIN_MAX
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_1"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_ciabatta"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              Ciabatta
            </Styled(Text)>
          </styled.div>
          <Styled(Text)
            className="option_amount "
          >
            $20.00
          </Styled(Text)>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_2"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_sourdough"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              Sourdough
            </Styled(Text)>
          </styled.div>
          <Styled(Text)
            className="option_amount "
          >
            $1.50
          </Styled(Text)>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_3"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_marblerye"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              MarbleRye
            </Styled(Text)>
          </styled.div>
          <Styled(Text)
            className="option_amount "
          >
            $1.50
          </Styled(Text)>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_4"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_multigrain"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              Multi Grain
            </Styled(Text)>
          </styled.div>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_5"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_wheatbaguette"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              Wheat Baguette
            </Styled(Text)>
          </styled.div>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_6"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_wheatwrap"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              Wheat wrap
            </Styled(Text)>
          </styled.div>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="option modifier_0ption_7"
      >
        <Styled(Flex)
          aria-checked={false}
          className="option_id_spinachwrap"
          onClick={[Function]}
          onKeyDown={[Function]}
          pl={
            Array [
              20,
              20,
              0,
            ]
          }
          role="radio"
          tabIndex={0}
        >
          <styled.div>
            <i
              className="fa fa-circle-thin"
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "inline-block",
                  "fontSize": "1rem",
                }
              }
            />
            <Styled(Text)
              className="option_desc "
            >
              Spinach wrap
            </Styled(Text)>
          </styled.div>
        </Styled(Flex)>
      </Styled(Flex)>
    </Styled(Flex)>
  </styled.div>
</ThemeProvider>
`;
